src/main/kotlin/com/openlattice/edm/processors/GetPartitionsFromEntitySetEntryProcessor.kt

```diff
@@ -5,7 +5,10 @@ import com.openlattice.rhizome.hazelcast.entryprocessors.AbstractReadOnlyRhizome
 import java.util.*
 
 class GetPartitionsFromEntitySetEntryProcessor : AbstractReadOnlyRhizomeEntryProcessor<UUID, EntitySet, Set<Int>>() {
-    override fun process(entry: MutableMap.MutableEntry<UUID, EntitySet>): Set<Int> {
-        return entry.value.partitions
+    override fun process(entry: MutableMap.MutableEntry<UUID, EntitySet?>): Set<Int> {
+        if ( entry.value == null ){
+            return setOf()
+        }
+        return entry.value!!.partitions
```

Comment:
I don't know Kotlin; just curious. Is the `!!` since you already check for `null` above?

Reply:
Ah, good catch. I just needed to assign entry.value to a variable and then its able to do a smart cast with it on line 12.
I was a little surprised that it can't detect this current code as not needing !!